activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskVariablesLocalCopiesIT.java

```diff
@@ -0,0 +1,150 @@
+import org.activiti.api.task.runtime.TaskRuntime;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@ContextConfiguration
+public class TaskVariablesLocalCopiesIT {
+
+    private static final String TWOTASK_PROCESS = "twoTaskProcess";
+
+    @Autowired
+    private ProcessRuntime processRuntime;
+
+    @Autowired
+    private TaskRuntime taskRuntime;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Before
+    public void init() {
+
+
+    }
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void shouldGetConfiguration() {
+        //when
+        ProcessRuntimeConfiguration configuration = processRuntime.configuration();
+
+        //then
+        assertThat(configuration).isNotNull();
+    }
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void shouldGetAvailableProcessDefinitionForTheGivenUser() {
+        //when
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+                                                                                                      50));
+        //then
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent())
+                .extracting(ProcessDefinition::getKey)
+                .contains(TWOTASK_PROCESS);
+    }
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void processInstanceVariablesCopiedIntoTasksByDefault() {
+
+        Map<String,Object> startVariables = new HashMap<>();
+        startVariables.put("start1","start1");
+        startVariables.put("start2","start2");
+
+        //when
+        ProcessInstance twoTaskInstance = processRuntime.start(ProcessPayloadBuilder.start()
+                .withProcessDefinitionKey(TWOTASK_PROCESS)
+                .withVariables(startVariables)
+                .build());
+
+        assertThat(processRuntime.variables(ProcessPayloadBuilder.variables().withProcessInstance(twoTaskInstance).build()))
+                .extracting("name", "value")
```

Comment:
Prefer method references to literal method name: `.extracting(VariableInstance::getName(), VariableInstance::getValue)`.